golang/pkg/events/events.go

```diff
@@ -1,24 +1,86 @@
 package events
 
 import (
+	"time"
+
+	"github.com/google/uuid"
 	"google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/types/known/anypb"
+	timestamppb "google.golang.org/protobuf/types/known/timestamppb"
 )
 
-func ToEvent(event proto.Message, source string, id string) ([]byte, error) {
+type cloudEventOptions struct {
+	id     *string
+	source *string
+	time   *time.Time
+}
+
+type Option = func(fields *cloudEventOptions)
+
+func WithId(id *string) Option {
+	return func(fields *cloudEventOptions) {
+		fields.id = id
+	}
+}
+
+func WithSource(source *string) Option {
+	return func(fields *cloudEventOptions) {
+		fields.source = source
+	}
+}
+
+func WithTime(time *time.Time) Option {
+	return func(fields *cloudEventOptions) {
+		fields.time = time
+	}
+}
+
+func ToEvent(event proto.Message, opts ...Option) ([]byte, error) {
+	fields := &cloudEventOptions{}
+	for _, opt := range opts {
+		opt(fields)
+	}
+
 	data, err := anypb.New(event)
 	if err != nil {
 		return nil, err
 	}
 
+	var defaultId string
```

Comment:
Do we need to declare this `defaultId` here?
Why not simply:
```
if fields.id == "" {
    fields.id = uuid.New().String()
}
```

The same for `source` and `time`

Reply:
Replied in the previous question